src/engine/UI/menu/Menu.cpp

```diff
@@ -75,4 +74,9 @@ void Menu::Render(const EngineContext &ctx) const {
     if (ctx.debug) {
         m_quadtree.DrawTree();
     }
+}
+
+void Menu::Update(const EngineContext &ctx) {
+    IDisplayLayer::Update(ctx);
+    RebuildQuadTree(ctx.windowSize);
```

Comment:
Do we need to rebuild the quadtree here again if we already  rebuild it in `IDisplayLayer::Update(..)`?